File: backend/app/schemas/project.py
from typing import Optional, List
from pydantic import Field
from app.schemas.base import BaseSchema, BaseDBModel

class ProjectBase(BaseSchema):
    title: str
    slug: str
    description: str
    industry: str
    client_name: str
    cover_image: str
    gallery_images: List[str] = Field(default_factory=list)
    technologies: List[str] = Field(default_factory=list)
    live_url: Optional[str] = None
    github_url: Optional[str] = None
    featured: bool = False
    display_order: int = 0

class ProjectCreate(ProjectBase):
    pass

class ProjectUpdate(BaseSchema):
    title: Optional[str] = None
    slug: Optional[str] = None
    description: Optional[str] = None
    industry: Optional[str] = None
    client_name: Optional[str] = None
    cover_image: Optional[str] = None
    gallery_images: Optional[List[str]] = None
    technologies: Optional[List[str]] = None
    live_url: Optional[str] = None
    github_url: Optional[str] = None
    featured: Optional[bool] = None
    display_order: Optional[int] = None

class ProjectResponse(ProjectBase, BaseDBModel):
    pass

File: src/types/project.ts
export interface Project {
  id: string
  title: string
  slug: string
  description: string
  industry: string
  client_name: string
  cover_image: string
  image?: string // fallback accessor
  gallery_images: string[]
  technologies: string[]
  tags?: string[] // mapped tags accessor
  category?: string // mapped category accessor
  live_url?: string
  github_url?: string
  featured: boolean
  display_order: number
  created_at?: string
  updated_at?: string
}
